components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java

```diff
@@ -161,4 +162,19 @@
      * @throws APIManagementException if failed to retrieve resources
      */
     List<UriTemplate> getAllResourcesForApi(String apiContext, String apiVersion) throws APIManagementException;
+
+    /**
+     * Get a list of APIs with given gateway labels and status
+     *@param gatewayLabels A list of gateway labels
+     * @param status Lifecycle status
+     * @throws APIManagementException If failed to get API list
+     */
+    List<API> getAPIsByStatus(List<String> gatewayLabels, String status) throws APIManagementException;
```

Comment:
Product release is different to component release. Component repos need to also follow the version strategy properly.